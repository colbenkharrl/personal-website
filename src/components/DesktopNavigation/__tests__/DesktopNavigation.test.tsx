import { render, screen } from '@testing-library/react';
import { DesktopNavigation } from '../';

jest.mock('../../NavItem', () => ({
  NavItem: () => <p>NavItem</p>,
}));

// mock useRouter
jest.mock('next/navigation', () => ({
  useRouter: jest.fn(),
  usePathname: jest.fn(),
}));

it('DesktopNavigation matches snapshot (regression test)', async () => {
  render(<DesktopNavigation data-testid="DesktopNavigation" />);

  expect(screen.getByTestId('DesktopNavigation')).toMatchSnapshot(
    'Full component snapshot.',
  );
});
