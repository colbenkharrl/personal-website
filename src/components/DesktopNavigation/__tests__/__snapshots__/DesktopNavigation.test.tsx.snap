// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DesktopNavigation matches snapshot (regression test): Full component snapshot. 1`] = `
<nav
  data-testid="DesktopNavigation"
>
  <ul
    class="flex rounded-full bg-white/90 px-3 text-sm font-medium text-zinc-800 shadow-lg shadow-zinc-800/5 ring-1 ring-zinc-900/5 backdrop-blur dark:bg-zinc-800/90 dark:text-zinc-200 dark:ring-white/10"
  >
    <p>
      NavItem
    </p>
    <p>
      NavItem
    </p>
    <p>
      NavItem
    </p>
    <p>
      NavItem
    </p>
    <p>
      NavItem
    </p>
  </ul>
</nav>
`;
